rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -174,16 +180,31 @@ case class PrettyPrinter(freeShift: Int, boundShift: Int) {
   private def buildPattern(patterns: Seq[Channel]): (Int, String) =
     ((0, "") /: patterns.zipWithIndex) {
       case ((patternsFree, string), (pattern, i)) =>
-        (patternsFree + freeCount(pattern), string + buildString(pattern) + {
-          if (i != patterns.length - 1) ", "
-          else ""
-        })
+        (patternsFree + freeCount(pattern),
+          string +
+            this
+              .copy(freeId = boundId, boundId = rotate(boundId))
+              .buildString(pattern) + {
+            if (i != patterns.length - 1) ", "
+            else ""
+          })
     }
 
   private def buildMatchCase(matchCase: MatchCase): String = {
     val patternFree: Int = matchCase.pattern.get.freeCount
-    PrettyPrinter(boundShift, 0).buildString(matchCase.pattern.get) + " => " +
-      PrettyPrinter(freeShift, boundShift + patternFree).buildString(matchCase.source.get)
+    "case " +
+      this
+        .copy(freeId = boundId, boundId = rotate(boundId), freeShift = boundShift, boundShift = 0)
+        .buildString(matchCase.pattern.get) + " => " +
+      this
+        .copy(boundShift = boundShift + patternFree)
+        .buildString(matchCase.source.get)
+  }
+
+  private def rotate(id: String): String = {
+    val newId = ((id.last + rotation - 97) % 26 + 97).toChar
```

Comment:
While this is moderately easier to read, it still conflates incrementing with rotation.
Just count in base 26 and then rotate on print. How hard is that?